### ISSUE TITLE
-s sources does not get processed for //DEPS

### ISSUE THREAD

**Issue by maxandersen** [jbangdev/jbang#1249]
**Describe the bug**
Have a file called `ec.jsh` that contains:

```
//DEPS org.eclipse.collections:eclipse-collections-api:11.0.0
//DEPS org.eclipse.collections:eclipse-collections:11.0.0

import org.eclipse.collections.api.*;
import org.eclipse.collections.api.bag.*;
import org.eclipse.collections.api.list.*;
```

run: `jbang -s ec.jsh -i` and it currently shows up with errors nor is the deps on the classpath.

**Comment by quintesse** [comment]
Hmm yeah. It's the job of the main source file to (recursively) collect all //-tags for itself and any/all `//SOURCES` it references.
In the case of simple tags like `//DEPS` we can simply at the end of that process add any dependencies that were added on the command line with `--deps`. But in the case of `--sources` it's not enough to simply add those files, we need to process each one of them, telling them to collect all their tags and then add _all_ information (sources, files, deps, etc) to the final result.

**Comment by quintesse** [comment]
Fixed by #1272 , closing
